### PR TITLE
[WIP] Pass configured format to share expiration picker

### PR DESCRIPTION
**Pull request by divergentdave**
In the files app, when you share a file or folder by link and set an expiration date, the format of the date changes to date-month-year if you edit the date. This patch makes the date picker use the same configured date format used elsewhere.

This isn't ready to merge yet, as I'm still setting up my testing environment and haven't actually tested the code yet.

Edit: After some testing, this clearly needs a lot more work, so that PHP either gets the same dd-mm-yyyy format date, or knows how to parse whatever format it gets passed.

### CODE REVIEW

**Comment by karlitschek** [comment]
@schiesbn

File: core/js/share.js [hunk 1]
```diff
@@ -842,7 +842,7 @@ OC.Share={
 		$('#expirationDate').slideDown(OC.menuSpeed);
 		$('#expirationDate').css('display','block');
 		$('#expirationDate').datepicker({
-			dateFormat : 'dd-mm-yy'
+			dateFormat : datepickerFormatDate
```
Comment (nickvergessen) [inline core/js/share.js]: Does not work like this for me:
```
DateTime::__construct(): Failed to parse time string (Jun./11/2015) at position 4 (/): Unexpected character
```

**Comment by nickvergessen** [comment]
Thanks for starting this, there seem to be some problems indeed:

Open the share dropdown it displayes: `2015-06-11 00:00:00` (YYYY-MM-DD and time?!)
Pick a date: `12-06-2015` (DD-MM-YYYY)

**Comment by nickvergessen** [comment]
@MorrisJobke if you have some time

**Comment by divergentdave** [comment]
FYI I pushed a new commit. This approach makes use of Datepicker's alternate field feature. The Datepicker updates a hidden input field with one date format, and the visible input field with another date format. This at least appears to work, with the cursory testing I've done.

**Comment by PVince81** [comment]
When I set the expiration date it now appears as "Jul./21/2015" which looks really weird.

I don't quite understand what the original issue is that you're trying to solve ?
Is the date format not consistent between code paths ?

**Comment by MorrisJobke** [comment]
I’m closing this pull request _for now_ because of inactivity.

@divergentdave we can always reopen it though! :) Please update the pull request and describe the issue detailed.
If you need any help, please ask on our [developer mailing list](http://mailman.owncloud.org/mailman/listinfo/devel) and join our IRC channel at #owncloud-dev (freenode network).

**Pull request closed without merging**